File: chat-service/src/main/java/com.stackroute.chatservice/service/ChatService.java
package com.stackroute.chatservice.service;

import com.stackroute.chatservice.model.Chat;
import com.stackroute.chatservice.model.ChatWithBuyer;
import com.stackroute.chatservice.model.MessageBody;
import org.springframework.stereotype.Service;


public interface ChatService {

    Chat saveChat(Chat chat) throws Exception;
    Chat editChatOfProduct(String buyerEmail,String loginedEmail,String message,Chat chat) throws Exception;
    Chat getChatByProduct(int productId) throws  Exception;
    ChatWithBuyer getChatByBuyerEmail(int productId, String buyerEmail) throws  Exception;
}

File: product-webapp/webapp/src/app/service/chat.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Chat } from '../model/chat';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ChatService {

  constructor(private http: HttpClient) { }

  baseUrl = environment.apiBaseUrl + '/chat-service';
  // baseUrl = "http://localhost:8099";


  saveChat(chat) {
    return this.http.post(this.baseUrl + "/api/v5/chat", chat);
  }

  editChat(buyerEmail, email, message, chat) {

    return this.http.post(this.baseUrl + "/api/v5/user/chats?buyerEmail=" + buyerEmail + "&loginedEmail=" + email + "&message=" + message, chat);
  }

  chatByProductId(productId) {
    return this.http.get(this.baseUrl + "/api/v5/chats/" + productId);
  }

  chatByProductIdAndBuyer(productId, loginedEmail) {
    return this.http.get(this.baseUrl + "/api/v5/user/chats?productId=" + productId + "&buyerEmail=" + loginedEmail);
  }

}
